### PR TITLE
AirbyteLib: Fix no-such-table-error

### LINKED ISSUE #30: 🐛 Bug: `NoSuchTableError` is raised if user tries to access a stream that did not receive records
Currently we only run 'create table' when and if record batches are being processed for that stream. Since we know which stream names are selected, and since we have the schema declared in the catalog for those streams, we could optionally _also_ create empty destination tables and initialize empty CachedDatasets for those streams.

This would be helpful because a user does not expect to get an error when checking for the results of a specific stream. Example, the "deployments" stream of the GitHub source is very likely to be empty. A user iterating through the streams they selected will get a confusing failure rather than just seeing an empty dataset.

Example error from Github (internal slack thread [here](https://airbytehq-team.slack.com/archives/C05DACCP6PM/p1707611362503629?thread_ts=1707603523.544589&cid=C05DACCP6PM)):
```
sqlalchemy.exc.NoSuchTableError: deployments
```

### PR DESCRIPTION
**Pull request by bindipankhudi**
Resolves: https://github.com/airbytehq/airbyte-lib-private-beta/issues/30

Note: (For AJ) The changes are same as what you reviewed earlier. Added some small fixes for linting tests to pass.  

**Main changes**
- processors.py: Added check for empty data streams in the process_input_stream method. This check adds a key for each empty stream to the data_batches dictionary.
- test_source_test_fixture.py: Added an empty stream (always_empty_stream) to source-test fixture and updated relevant tests. 

**Testing**
All tests for airbyte-lib pass locally. However doc generation tests are failing here. :(

### CODE REVIEW

File: airbyte-lib/airbyte_lib/_processors.py [hunk 1]
```diff
@@ -45,6 +45,7 @@
 
 
 DEFAULT_BATCH_SIZE = 10_000
+DEBUG_MODE = False  # Set to True to enable additional debug logging.
```
Comment (aaronsteers) [inline airbyte-lib/airbyte_lib/_processors.py:48]: Nothing we need to change here, but later we might consider optionally having a single DEBUG_MODE flag, perhaps in a shared module like `airbyte_lib._constants` or `airbyte_lib.constants`.

File: airbyte-lib/pyproject.toml [hunk 2]
```diff
@@ -50,6 +50,7 @@ types-requests = "2.31.0.4"
 freezegun = "^1.4.0"
 airbyte-source-faker = "^6.0.0"
 viztracer = "^0.16.2"
+tomli = "^2.0"
```
Comment (aaronsteers) [inline airbyte-lib/pyproject.toml:53]: Is `tomli` used? I couldn't see references to it, so I am guessing perhaps it was used in a previous version? If not needed, we can delete.

File: airbyte-lib/pyproject.toml [hunk 3]
```diff
@@ -63,6 +64,7 @@ markers = [
 
 [tool.ruff.pylint]
 max-args = 8  # Relaxed from default of 5
+max-branches = 15 # Relaxed from default of 12
```
Comment (aaronsteers) [inline airbyte-lib/pyproject.toml:67]: Makes sense 👍

File: airbyte-lib/tests/integration_tests/test_source_test_fixture.py [hunk 4]
```diff
@@ -682,11 +707,15 @@ def test_sync_to_postgres(new_pg_cache_config: PostgresCacheConfig, expected_tes
...
-            result[stream_name].to_pandas(),
-            pd.DataFrame(expected_data),
-            check_dtype=False,
-        )
+        if len(cache[stream_name]) > 0:
+            pd.testing.assert_frame_equal(
+                result[stream_name].to_pandas(),
+                pd.DataFrame(expected_data),
+                check_dtype=False,
+            )
```
Comment (aaronsteers) [inline airbyte-lib/tests/integration_tests/test_source_test_fixture.py:715]: For my info, does `assert_frame_equal()` not work with an empty dataset?

**Review by aaronsteers — APPROVED** [review]
A couple questions and comments but nothing blocking (except to confirm we need `tomli`).

Merge when ready 🚀 :shipit:

File: airbyte-lib/tests/integration_tests/test_source_test_fixture.py (hunk 4 above)
Comment (bindipankhudi) [reply airbyte-lib/tests/integration_tests/test_source_test_fixture.py:715]: nope, trying to run .to_pandas() on an empty dataset fails, not meant to work apparently.

Comment (aaronsteers) [reply airbyte-lib/tests/integration_tests/test_source_test_fixture.py:715]: Makes sense! Thanks for confirming. 👍

**Review by aaronsteers — APPROVED** [review]

**Pull request merged**